comp/forwarder/defaultforwarder/worker.go

```diff
@@ -150,6 +150,14 @@ func (w *Worker) ScheduleConnectionReset() {
 // callProcess will process a transaction and cancel it if we need to stop the
 // worker.
 func (w *Worker) callProcess(t transaction.Transaction) error {
+	requeue := func() {
+		select {
+		case w.RequeueChan <- t:
+		default:
+			w.log.Errorf("dropping transaction because the retry goroutine is too busy to handle another one")
```

Comment:
Given that the retry goroutine is likely already by the time we hit this (`stopRetry` happens before `w.Stop`), perhaps something like this would be clearer:

```suggestion
			w.log.Errorf("dropping transaction because the retry buffer is full while stopping")
```

Reply:
Just to be clear, this is an existing message that I moved here. I'm still happy to change it if we think that's best, but wanted to double check 😄

Reply:
> Just to be clear, this is an existing message that I moved here.

Did you mean to remove it from the old place then? Because I think it is still part of `process`.

But there it happens during normal operation, when retry goroutine is handling the queue. But this isn't the case here, when it is called on stop. I think it would be clearer if the message matched the state of things.

Reply:
Oh, yep, you're correct. I've just pushed a commit that actually extracts the function rather than copying it, which is what I intended.